candi/bashible/common-steps/node-group/003_discover_node_ip.sh.tpl

```diff
@@ -34,14 +34,25 @@ fi
 {{- end }}
 
 {{- if eq .nodeGroup.nodeType "Static" }}
-  {{- if not (hasKey .nodeGroup "static") }}
-    >&2 echo "ERROR: nodeGroup.static.internalNetworkCIDRs must exist for static node"
-    exit 1
-  {{- else if not (hasKey .nodeGroup.static "internalNetworkCIDRs") }}
-    >&2 echo "ERROR: nodeGroup.static.internalNetworkCIDRs must exist for static node"
+ip_route_get_cmd="ip route show scope link proto kernel"
+  {{- if and (hasKey .nodeGroup "static") (hasKey .nodeGroup.static "internalNetworkCIDRs")}}
+internal_network_cidrs={{ .nodeGroup.static.internalNetworkCIDRs | join " " | quote }}
+  {{- end }}
+
+if [[ -z "$internal_network_cidrs" ]]; then
+  # if internal network cidrs is not set, and the node has one interface, use its network as internal_network_cidr
+  if [[ "$($ip_route_get_cmd | wc -l)" -eq 1 ]]; then
+    internal_network_cidrs="$($ip_route_get_cmd | awk '{print $1}')"
+  else
+    bb-log-error "Cannot discover internal network CIDRs. Node has more than one interface, and nodeGroup.static.internalNetworkCIDRs is not set."
```

Comment:
The `nodeGroup.static.internalNetworkCIDRs` property is an internal. Users should not have access to this. They should deploy the StaticClusterConfiguration to the cluster instead.

Reply:
Changed, thanks